tests/xdg_toplevel_stable.cpp

```diff
@@ -431,6 +439,51 @@ TEST_F(XdgToplevelStableTest, null_parent_can_be_set)
     client.roundtrip();
 }
 
+TEST_F(XdgToplevelStableTest, when_parent_is_set_to_self_error_is_raised)
+{
+    wlcs::Client client{the_server()};
+    ConfigurationWindow window{client};
+    xdg_toplevel_set_parent(window, window);
+    wl_surface_commit(window);
+    try
+    {
+        client.roundtrip();
+    }
+    catch (wlcs::ProtocolError const& err)
+    {
+        return;
```

Comment:
Only if the protocol version is new enough to support `xdg_toplevel::invalid_parent`. We could test this, but I really don't see why it matters.

Reply:
Technically this shouldn't *be* a protocol error until `xdg-shell` v5. A truly pedantic test would bind to `xdg-shell` < v5, set up a parent loop, and assert no protocol error. :wink:

It's not a big thing (because protocol errors can't be used for error recovery), but it can be helpful for client developers to send the specified error. As an aspirational conformance suite, we should be checking adherence to the protocol; that specifies a specific error, it's cheap to check it, we should.

Reply:
(I don't expect you to write the “parent loops are fine on <v5” test, but we *should* gate this test on ≥v5, and test the correct error is sent)